common/modal.js

```diff
@@ -49,6 +49,9 @@
             vm.clickActionMessage =  messageMap.recordAvailabilityError.multipleRecords;
         } else if(vm.params.errorCode == 'Record Not Found'){
             vm.clickActionMessage = messageMap.recordAvailabilityError.noRecordsFound;
+            if(params && !params.isLoggedIn){
+              params.message = 'No matching record found for the given filter or facet.<br>You may not be authorized to view this record.';
```

Comment:
First, this should be factored out into the message object.

Second, the first part of this message is already a message in `errorMessages`. The second part is part of a message in `messageMap` (`messageMap.unauthorizedMessage`). That `unauthorizedMessage` key/value pair should probably be split into 2 separate messages so that `unauthorizedMessage` is only `You are not authorized to perform this action.`.  And a new key/value pair is defined with the second part of the string, `Please report this problem to your system administrators.`.